lib/rubocop/cop/rails/timeout_from_transaction.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for transactions wrapped in timeouts,
+      # motivated by change in Rails behavior that will
+      # cause exited transactions to be rolled back
+      # rather than committed (pre ActiveRecord 7 behavior).
+      #
+      # @example
+      #   # bad
+      #   Timeout.timeout(10) do
+      #     ApplicationRecord.transaction do
+      #       user.save
+      #     end
+      #   end
+      class TimeoutFromTransaction < Base
```

Comment:
I'm not sure about the better name... what about the following name like this?

```suggestion
      class TransactionInTimeout < Base
```

Reply:
sure, I'm fine with the name change.